support-frontend/assets/helpers/subscriptionsForms/subscriptionCheckoutReducer.ts

```diff
@@ -12,33 +12,38 @@ import { addressReducerFor } from 'components/subscriptionCheckouts/address/addr
 import type { Csrf as CsrfState } from 'helpers/csrf/csrfReducer';
 import csrf from 'helpers/csrf/csrfReducer';
 import type { IsoCountry } from 'helpers/internationalisation/country';
-import type { ReduxState } from 'helpers/page/page';
 import type { BillingPeriod } from 'helpers/productPrice/billingPeriods';
 import type { FulfilmentOptions } from 'helpers/productPrice/fulfilmentOptions';
 import type { ProductOptions } from 'helpers/productPrice/productOptions';
 import type { SubscriptionProduct } from 'helpers/productPrice/subscriptions';
+import type { CommonState } from 'helpers/redux/commonState/state';
 import type { FormState } from 'helpers/subscriptionsForms/formFields';
 import { createFormReducer } from 'helpers/subscriptionsForms/formReducer';
 import type { Option } from 'helpers/types/option';
 import type { User } from 'helpers/user/userReducer';
 import { createUserReducer } from 'helpers/user/userReducer';
 
+export type ReduxState<PageState> = {
+	common: CommonState;
+	page: PageState;
+};
+
 export type CheckoutState = ReduxState<{
 	checkout: FormState;
 	user: User;
 	csrf: CsrfState;
 	marketingConsent: MarketingConsentState;
 	billingAddress: AddressState;
-	hasDeliveryAddress: false;
+	deliveryAddress?: AddressState;
 	directDebit: DirectDebitState;
 }>;
+
 export type WithDeliveryCheckoutState = ReduxState<{
 	checkout: FormState;
 	csrf: CsrfState;
 	marketingConsent: MarketingConsentState;
 	billingAddress: AddressState;
 	deliveryAddress: AddressState;
-	hasDeliveryAddress: true;
```

Comment:
`hasDeliveryAddress` was not being read anywhere in the code, and created an unnecessary type conflict between these two state variations.